tests/integration/test_workflow_panels_happy_path.py

```diff
@@ -0,0 +1,1351 @@
+            "qtpy.QtWidgets.QMessageBox.warning",
+            lambda *args, **kwargs: pytest.fail(
+                "WARNING messagebox:\n" + f"    args: {args}\n" + f"    kwargs: {kwargs}", pytrace=False
+            ),
+        )
+        self._warningMessageBox.start()
+
+        self._criticalMessageBox = mock.patch(
+            "qtpy.QtWidgets.QMessageBox.critical",
+            lambda *args, **kwargs: pytest.fail(
+                "CRITICAL messagebox:\n" + f"    args: {args}\n" + f"    kwargs: {kwargs}", pytrace=False
+            ),
+        )
+        self._criticalMessageBox.start()
+
+        # patch log-warnings QMessage box: runs using `QMessageBox.exec`.
+        self._logWarningsMessageBox = mock.patch(
+            "qtpy.QtWidgets.QMessageBox.exec",
+            lambda self, *args, **kwargs: QMessageBox.Ok
+            if (
+                "The backend has encountered warning(s)" in self.text()
+                and "InstrumentDonor will only be used if GroupingFilename is in XML format." in self.detailedText()
+            )
+            else pytest.fail(
+                "unexpected QMessageBox.exec:\n"
+                + f"    args: {args}\n"
+                + f"    kwargs: {kwargs}\n"
+                + f"    text: '{self.text()}'\n"
+                + f"    detailed text: '{self.detailedText()}'",
+                pytrace=False,
+            ),
+        )
+        self._logWarningsMessageBox.start()
+
+        # Automatically continue at the end of each workflow.
+        self._actionPrompt = mock.patch(
+            "snapred.ui.presenter.WorkflowPresenter.ActionPrompt.prompt",
+            lambda *args: TestGUIPanels._actionPromptContinue(
+                *args, match=r".*The workflow has been completed successfully.*"
+            ),
+        )
+        self._actionPrompt.start()
+        # ---------------------------------------------------------------------------
+
+        with Resource.open("../../src/snapred/resources/style.qss", "r") as styleSheet:
+            qapp.setStyleSheet(styleSheet.read())
+
+        # Establish context for each test: these normally run as part of `src/snapred/__main__.py`.
+        self.exitStack = ExitStack()
+        self.exitStack.enter_context(amend_config(data_dir=_prepend_datasearch_directories(), prepend_datadir=True))
+        yield
+
+        # teardown...
+        self._warningMessageBox.stop()
+        self._criticalMessageBox.stop()
+        self._logWarningsMessageBox.stop()
+        self._actionPrompt.stop()
+        self.exitStack.close()
+
+    @staticmethod
+    def _actionPromptContinue(title, message, action, parent=None, match=r".*"):  # noqa: ARG004
+        _pattern = re.compile(match)
+        if not _pattern.match(message):
+            pytest.fail(
+                f"unexpected: ActionPrompt.prompt('{title}', '{message}'...)\n"
+                + f"    expecting:  ActionPrompt.prompt(...'{message}'...)"
+            )
+
+    @pytest.mark.integration()
```

Comment:
Since you have a marker at the class level, the marker here is not required

Reply:
I'll add this change.